sdk/cosmos/azure-cosmos-spark_3-0_2-12/pom.xml

```diff
@@ -0,0 +1,338 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-client-sdk-parent</artifactId>
+    <version>1.7.0</version> <!-- {x-version-update;com.azure:azure-client-sdk-parent;current} -->
+    <relativePath>../../parents/azure-client-sdk-parent</relativePath>
+  </parent>
+
+  <groupId>com.azure</groupId>
+  <artifactId>azure-cosmos-spark_3-0_2-12</artifactId>
```

Comment:
Why do we need the suffix `_3-0_2_12`?

Reply:
Suffix is needed because if we ever want to support the Spark connector for different (major/breaking) Spark versions we will need different binary packages. Because we would need to actively support multiple versions for some period of time having the versions of the main dependencies Spark and Scala is a way to distinguish the different packages (and source code). This is pretty common practice in Spark - especially for connectors.

Reply:
Elsewhere where we have encountered similar concerns (i.e. Spring) we have determined that the Maven classifier functionality is sufficient to allow users to differentiate between releases. This is documented in the Spring guidelines [here](https://azure.github.io/azure-sdk/java_spring.html#java-spring-classifiers), for reference. It would be nice if you could do the same, because I always think this style of 'multiple libraries to support the same conceptual product' just leads to a lot of low-value noise in the com.azure group ID.

Reply:
I agree that it makes sense to look at what other communities have identified as the appropriate pattern - and wouldn't recommend following this pattern in Spring. But in the Spark eco-system the community has clearly aligned on the pattern to have separate packages and artifactIds for different runtime versions (often with just the Spark runtime version because the Scala version is kind of defined by it automatically, sometime like here with the scala version). I am fine with removing the Scala version as well - so making this azure-cosmos-spark_3-0 if that addresses some of your concerns. But because removing the version form the package/artifact Id is contrary to well adopted patterns within the Spark community I think that is a bad idea.
Are you insisting on removing the the Scala version - or is either a hard no-go for you?

Reply:
In the Spark community you can basically find three different patterns:
1) Keeping at least Scala version - or also Spark runtime version - in the artifactId
2) Using Major+Minor versions in your connector aligning with the correlating Spark Major+Minor version
3) Using Spark+Scala as provided references documenting which versions the connector works in

Our preference is clearly for 1) because managing dependencies in Spark can be very painful - setting expectations which runtime version a certain connector supports is critical - we also document out-of-band what the recommended versions of the connector are for a certain Databricks or Synapse runtime (the runtimes the vast majority of our customers are using) - but the more explicit the better.
The main problem is that you need to manage a two-dimensional semantic versioning - any breaking changes (or new features) in you own connector code - for example config changes, adding new features like AAD authentication etc. require a major and/or minor version bump of the connector - possibly across multiple versions of runtimes. As a customer you build your application against a certain version of the connector (often Spark "application" are basically Jupyter Notebooks) - the dependencies like Spark connectors available are managed on the cluster level by the admins. Being explicit in the dependencies between runtimes supported and application/connector version helps to bridge the perspectives of the Devs and admins who often manage clusters as shared resources.

Samples for naming pattern of the artifactId of other Spark connectors which don't ship built-in within Spark but externally:
Cassandra: https://mvnrepository.com/artifact/com.datastax.spark/spark-cassandra-connector
CouchBase: https://mvnrepository.com/artifact/com.couchbase.client/spark-connector_2.11

Reply:
Moved the spark library to its own group Id com.azure.cosmos.spark

So the concerns around number of artifacts is addressed. Also switching to only adding the scala version to the artifactId (not also the Spark version) to align with the pattern Cassandra, Kafka and Mongo are using.